internal/cluster/cluster.go

```diff
@@ -207,7 +207,22 @@ func (m *Manager) DeregisterCluster(ctx context.Context, c *common.Cluster) erro
 		return metricsErr
 	}
 
-	err := m.registrationAPI.DeregisterCluster(ctx, c)
+	// Delete discovery image for deregistered cluster
+	discoveryImage := fmt.Sprintf(s3wrapper.DiscoveryImageTemplate, c.ID.String())
+	exists, err := m.objectHandler.DoesObjectExist(ctx, discoveryImage)
+	if err != nil {
+		m.log.WithError(err).Errorf("Failed to find cluster discovery image %s in s3", discoveryImage)
```

Comment:
```suggestion
		m.log.WithError(err).Errorf("Failed to find cluster discovery image %s", discoveryImage)
```

- Is it expected that for every cluster there must be a discovery image? 
- If we are deregistering the cluster. Does it matter that there was no discovery image?

Reply:
The discovery-image ISO will be created when the user asks to generate the ISO.
Therefore if the cluster was just created, but 'Generate ISO' wasn't invoked, there won't be a discovery image for the cluster.

Deregistering a cluster means the cluster should be deleted. It will not appear in the UI and will not be available for the user. Therefore not having the discovery-image shouldn't matter.